xbmc/Application.cpp

```diff
@@ -4865,6 +4865,13 @@ bool CApplication::OnMessage(CGUIMessage& message)
       if (url.IsProtocol("plugin"))
         XFILE::CPluginDirectory::GetPluginResult(url.Get(), file);
 
+      // Don't queue if next media type is different from current one
+      if ((!file.IsVideo() && m_pPlayer->IsPlayingVideo()) ||
```

Comment:
>  Any reason why you don't stop it explicitly?

I guess you mean using "OnNothingToQueueNotify()." Honestly, I just didn't think about it, but seems like a much better solution. I'll test it and come back with a new patch.

> (btw: our coding style defines curly brackets on next line.)

I'll fix it, too.